weight_data_dialog.py

```diff
@@ -89,12 +94,38 @@ def __init__(self, iface, project_definition):
         self.frame.javaScriptWindowObjectCleared.connect(self.setup_js)
         self.web_view.loadFinished.connect(self.show_tree)
         self.json_updated.connect(self.handle_json_updated)
+        self.populate_style_by_field_cbx()
 
+    def update_project_definition(self, project_definition):
+        self.project_definition = deepcopy(project_definition)
+        self.populate_style_by_field_cbx()
+
+    def set_active_layer_numeric_fields(self):
         self.active_layer_numeric_fields = [
             field.name()
-            for field in iface.activeLayer().dataProvider().fields()
+            for field in self.iface.activeLayer().dataProvider().fields()
             if field.typeName() in NUMERIC_FIELD_TYPES]
 
+    def populate_style_by_field_cbx(self):
+        self.set_active_layer_numeric_fields()
+        # the empty list will be filled by get_field_names
+        fields_in_proj_def = get_field_names(self.project_definition, [])
+        fields_for_styling = [
+            field for field in self.active_layer_numeric_fields
+            if field in fields_in_proj_def]
+        # block signals to avoid performing the onchange actions while adding
+        # items programmatically
+        self.ui.style_by_field_cbx.blockSignals(True)
```

Comment:
I can't reuse that, because I need to get only the fields that are both in the layer and in the project definition